packages/dataviews/src/dataform.tsx

```diff
@@ -0,0 +1,106 @@
+/**
+ * External dependencies
+ */
+import type { Dispatch, SetStateAction } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { TextControl } from '@wordpress/components';
+import { useCallback, useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import type { Form, Field, NormalizedField } from './types';
+import { normalizeFields } from './normalize-fields';
+
+type DataFormProps< Item > = {
+	data: Item;
+	fields: Field< Item >[];
+	form: Form;
+	onUpdate: Dispatch< SetStateAction< Item > >;
```

Comment:
In DataViews, we have `onChangeView` and `onSelectionChange` and here we have `onUpdate`

I think I'd love for these to be:

 - onChangeView
 - onChangeSelection
 - onChange (no suffix here to match Controls and because it's slightly different in terms of semantic, we're changing the data)

I'm also wondering whether `data` should be just `value` like the Controls component but that's more arguable.

Reply:
Renamed `onUpdate` to `onChange` at https://github.com/WordPress/gutenberg/pull/63032/commits/374702023d68c87eee292d4134af19b622728938

`data` is in line with `DataViews` and also at some point we should work with multiple items, so I lean towards keeping it.

Reply:
https://github.com/WordPress/gutenberg/pull/63087 renames `onSelectionChange` to `onChangeSelection` for dataviews.